pages/checkout/terms.tsx

```diff
@@ -49,83 +61,132 @@ const InfoBox = styled.div`
   margin-bottom: ${tokens.spacings.comfortable.medium}
 `
 
+const DataSourceErrorContainer = styled.div`
+  > *:not(:last-child) {
+    margin-bottom: ${tokens.spacings.comfortable.medium};
+  }
+`
+
 type Props = {
   asset?: Collibra.Asset | null
+  error?: string
   rightsToUse?: {
     name: string
     value: string
   }
 }
 
-const CheckoutTermsView: NextPage<Props> = ({ asset, rightsToUse }) => {
+const CheckoutTermsView: NextPage<Props> = ({ asset, error, rightsToUse }) => {
   const intl = useIntl()
   const router = useRouter()
-  const [error, setError] = useState(false)
+  const [formError, setFormError] = useState(false)
   const { checkoutData, setCheckoutData } = useCheckoutData()
 
   const hasAcceptedTerms = checkoutData?.terms?.termsAccepted
-  const onContinue = () => {
-    if (hasAcceptedTerms) {
-      setError(false)
-      router.push({
-        pathname: "/checkout/access",
-        query: { id: asset?.id },
+
+  useEffect(() => {
+    if (asset) {
+      setCheckoutData({ asset: { id: asset.id, name: asset.name } })
```

Comment:
This will be updated twice in development, shouldn't we make it pure like ...asset 🤔

Reply:
that is true

Reply:
although, `asset` contains a lot more than just `name` and `id`, so it'd have to be `...(checkoutData.asset ?? {})`

Reply:
Yup, whatever it takes to make it pure 😄